### ISSUE TITLE
Undefined method `slug=' but the column exists

### ISSUE THREAD

**Issue by PavelJ** [norman/friendly_id#739]
Hi,
I have configured frindly id according to the documentation but I have ecountered an error when I am trying to initialize my database. When I try to save a model, i just get the following error:
**NoMethodError: undefined method `slug=' for #Hotel:0x3d19a60**
- **Rails version:** 4.2.4
- **Ruby version:** 2.1.7
- **Frienldy ID version:** 5.1.0

Part of my Hotel model:

```
class Hotel < ActiveRecord::Base
  extend FriendlyId
  friendly_id :generate_slug, use: :slugged

  def generate_slug
    "#{self.name} #{self.id}"
  end

  def should_generate_new_friendly_id?
    name_changed? || new_record?
  end

  def normalize_friendly_id(text)
    super(I18n.transliterate(text))
  end
```

Part of initialization code, where the error is thrown:

```
rest = Hotel.new(:name => "Kamenný stul", :telephone => "777671105 ", :email => "info@kamennystul.cz", :web_page => "www.kamennystul.cz", :state => 1, :city_search => "Praha,Praha 1", :smoking => 1,
                      :address_attributes => {:city => "Praha 1", :street => "Staromestské námestí 17/551", :postal_code => "11000", :country_code => "CZ"})
  ObjectSkipValidator.setup_validations(rest, nil)
  rest.save!
```

Of course the table has slug column, my migration:

```
AFFECTED_TABLES = [:restaurants, :hotels]

  def change
    AFFECTED_TABLES.each do |tab|
      add_column tab, :slug, :string
      add_index tab, :slug, unique: true   
    end
  end
```

And friendly id created this migration:

```
class CreateFriendlyIdSlugs < ActiveRecord::Migration
  def change
    create_table :friendly_id_slugs do |t|
      t.string   :slug,           :null => false
      t.integer  :sluggable_id,   :null => false
      t.string   :sluggable_type, :limit => 50
      t.string   :scope
      t.datetime :created_at
    end
    add_index :friendly_id_slugs, :sluggable_id
    add_index :friendly_id_slugs, [:slug, :sluggable_type]
    add_index :friendly_id_slugs, [:slug, :sluggable_type, :scope], :unique => true
    add_index :friendly_id_slugs, :sluggable_type
  end
end
```

I can see the column in the database. Rake task is started only in development environment using RAILS_ENV=development. I don't see any error with my setup and I have tried searching for some solution, but it just looks like a bug.
Also if I just try to create attr_accessor, task succesfully ends, but there are no slugs in the database. I have also tried dropping the whole database and recreate it from the beginning.

**EDIT:**
It appears, that Rails didn't load the column into attributes hash in ActiveModel::AttributeMethods so when attribute_missing is called, "slug" is not found....still trying to figure it out.

**Comment by PavelJ** [comment]
At the end the problem was with a rake task itself, migrations weren't handled properly, don't know exactly why. Sorry for bothering, this issue can be closed - it is not related to friendly_id at all.

**Comment by zedtux** [comment]
Well I fall in the same case. On my dev box all is working fine, but on staging I got an error during the migration, and then the `slug` column was present already when I re-started the migration (which shouldn't be as Rails should have done the rollback).

Anyway, I have deleted manually the column and ran again the migration and all went well. Then a page of the app wasn't working and looking at the log I got the error `undefined method `slug' for ...` which is weird as when I load the last object, I can see the column with a value.

I have then restarted the docker containers of my Rails app, then it worked like if Rails has reloaded the columns of that model...

It could be that my issue running the migration is the reason of this strange behaviour but it's also strange that I wasn't the only one.
